### ISSUE TITLE
Import Pod: Markdown does not work on Windows

### ISSUE THREAD

**Issue by SR--** [dendronhq/dendron#848]
## Please select the area your Bug applies to. (Multiple selections are Possible)

- [x] Onboard. Initial vault setup and import/export pods 
- [ ] Create. Note creation, lookup, snippets and templates
- [ ] Retrieve. Backlinks, references, graph view
- [ ] Structure. Refactoring, multi-vault and schemas
- [ ] Publish. Sharing your repo with the world
- [ ] Misc (Choose this if your not sure)

## Describe the bug

Import Pod: Markdown command does not complete on Windows. Probably because if incorrect path interpretation.

## To Reproduce

1. Try to import
2. The process does not complete

## Expected behavior

Notes imported.

## Desktop (please complete the following information)

- OS: Windows 10
- Version 47

**Comment by kpathakota** [comment]
@SR-- could you provide more screenshots to show how it's failing? Might just be a path issue. I'm blocked on repro'ing because I keep getting "The configuration is invalid YAML" even though I'm only changing a few things. We'll investigate further.

**Comment by SR--** [comment]
In my case, the import did not match the wiki links int he incoming notes to the newly derived file names that used the dot separators to represent former folders. But this is a separate issue. Maybe a missing feature?

**Comment by jonathanyeung** [comment]
Hi @SR--, we made a fix such that if a particular note had an import issue, we'd go ahead and import the rest and just report the note(s) that had issues.  For the two files that were missing, is there anything special about the formatting?  We've seen issues before where if a note with frontmatter at the top of the note had invalid YAML, then it'll cause an issue. 

> the import did not match the wiki links int he incoming notes

For this, could you provide an example of a wiki-link that failed to convert? The autoconversion to . format _should_ be supported.


Thanks for helping test this and appreciate the patience!

**Comment by SR--** [comment]
Most of note imported in v48, apart from two that probably did have nonstandard frontmatter. I deleted that, added them by hand and run the Doctor.

On the links. None of the links in the form [[blah]] got updated to read [[blah.blah]] when the files got renamed into blah.blah.md to represent the incoming folder hierarchies. In my case, the links to just first attempts so I am renaming them by hand as I go, but for working collections it would be an issue.

**Comment by kpathakota** [comment]
Reading through this thread, I *think* there's nothing else to do here. @SR-- / @jonathanyeung  - let me know if that's not the case and you expect further action. For now, I've added a tag that will auto-close the issue in 7 days if there's nothing else required here.

**Comment by jonathanyeung** [comment]
Thanks SR-- for the help on this issue. 

Since the major import issue was fixed (https://github.com/dendronhq/dendron/pull/865) and we now report issues per-file that had import issues, I'll close this issue. 

I'm unable to repro the wikilink [[blah]] issue; conversion from folder to dot notation should be supported. If this remains a problem, let's track that with a different issue.

**Comment by SR--** [comment]
I did not check it. But the folder conversion did not work for me. I just chose to fix that by hand. Maybe it does not work if there are problem notes. Breaking frontmatter in a couple to see what happens with the rest might be a way to check......
________________________________

**Comment by voothi** [comment]
@SR-- 
Hi! Looks like I have a similar problem. After Pod Markdown Import on Windows (on other OS, I did not check) Wikilinks do not change on new, in line with the new hierarchy of files. 
I recorded video on this problem 
https://youtu.be/DJBh-WlXsRg
Please say that my problem coincides with your or not.

**Comment by SR--** [comment]
Yes, looks similar. I have refactored all the incoming notes since and could only check with simulated files. But none of the link got updated to [[something.somethingelse]] format to match file names.

**Comment by voothi** [comment]
Ok, thanks, then for me this problem is also relevant! I would like to go to Dendron as soon as possible.

**Comment by voothi** [comment]
Hello, I tested on Linux with the same version of VSCode and Dendron. Behavior is the same as on video from Windows https://youtu.be/djbh-wlxsrg
Perhaps I do not understand how it should work.

![image](https://user-images.githubusercontent.com/9587340/126568223-1aaf0caa-3070-43b7-8e73-833e132d980b.png)

**Comment by jonathanyeung** [comment]
Hi @voothi - looking at the video, it seems that the wikilinks are pointing to files that are in different folders, but the wikilink contents only contain the note name itself and does not contain a path prefix.  Is this the case? 

Dendron import doesn't support this - if there is a path prefix in the wikilink, then it should get converted to Dendron's dot notation. 

Do you have some other extension installed in your VS Code that allows the wikilink to a note in a different folder to work?

**Comment by voothi** [comment]
Hi, @jonathanyeung! 

> Hi @voothi - looking at the video, it seems that the wikilinks are pointing to files that are in different folders, but the wikilink contents only contain the note name itself and does not contain a path prefix. Is this the case?
> 
Yes, files store in different folders. 
My current notebook has such a structruct:
d - dir
f - file
```
d H:\Dendron_Test_Pod_Import_Md\Markdown_Notes
f H:\Dendron_Test_Pod_Import_Md\Markdown_Notes\20210719130042_Main.md
...
d H:\Dendron_Test_Pod_Import_Md\Markdown_Notes\20210719130143_Computer
f H:\Dendron_Test_Pod_Import_Md\Markdown_Notes\20210719130143_Computer.md
f H:\Dendron_Test_Pod_Import_Md\Markdown_Notes\20210719130224_Computer_Mem.md
..
```

> Dendron import doesn't support this - if there is a path prefix in the wikilink, then it should get converted to Dendron's dot notation.
> 
Ok, but after markdown import I get renamed notes with old broken wikilinks.


> Do you have some other extension installed in your VS Code that allows the wikilink to a note in a different folder to work?
Yes, I use https://marketplace.visualstudio.com/items?itemName=kortina.vscode-markdown-notes in other VSCode instance on same PC.
I don't use Obsidian.md for regular work. Sometimes I use her only for search.
I did know about specific notation in Obsidian.md. Thanks for info. 
I read earlier about this https://wiki.dendron.so/notes/f23a6290-2dec-45dc-b616-c218ee53db6b/#links

I solve my problem in split import subdir as individual vault. This is correct approach?

Thanks.

**Comment by Harshita-mindfire** [comment]
Thanks @voothi  for the help on this issue.

With the latest release of Dendron(v0.68), The wikilinks for notes in same directory level will now be updated. #1615
Updating the wikilinks while import has been already fixed with #1607, Let's track the remaining markdown issue with #1608.

**Comment by voothi** [comment]
@Harshita-mindfire 
Hi! 
I checked this issue on v0.69.0. 
Looks good,
but 
this don't work still 
![image](https://user-images.githubusercontent.com/9587340/142049400-718f4cde-0775-4fad-861f-b46233c480f9.png)
Thanks!
Please, reopen this issue.

**Comment by Harshita-mindfire** [comment]
Hey @voothi , we currently don't support resolving ambiguous wikilinks. The reason being the difficulty in appending the prefix. Taking one of the above examples as context,  the wikilink `[[20210719130312_CS_Algorithm]]` in `20210719130042_Main.md`  can be present  in either or both directories( `20210719130303_CS` , `20210719130143_Computer` ) making it difficult to resolve.

**Comment by voothi** [comment]
Hey @Harshita-mindfire, I seems catch it! Indeed, it is ok. 
Thanks for good work!
